x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/runtime_mappings/runtime_mappings_editor.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEqual } from 'lodash';
+import React, { memo, FC } from 'react';
+import { EuiCodeEditor } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isRuntimeMappings } from '../../../../../../../common/util/runtime_field_utils';
+
+interface Props {
+  convertToJson: (data: string) => string;
+  setAdvancedRuntimeMappingsConfig: React.Dispatch<string>;
+  setIsRuntimeMappingsEditorApplyButtonEnabled: React.Dispatch<React.SetStateAction<boolean>>;
+  advancedEditorRuntimeMappingsLastApplied: string | undefined;
+  advancedRuntimeMappingsConfig: string;
+  xJsonMode: any;
```

Comment:
I'm not sure if you need to provide it via props. I think you should initialize it in this component instead.
```typescript
const xJsonMode = useMemo(() => new XJsonMode(), []);
```

Reply:
Good question - It's needed in the parent component for when we want to set initial runtimeMappings on first load - like for index patterns with runtime fields or when cloning a job.

Reply:
can we provide a type instead of `any`? :) I assume you can use `XJsonMode`